sycl/include/CL/sycl/types.hpp

```diff
@@ -529,11 +529,16 @@ template <typename Type, int NumElements> class vec {
                               std::is_integral<DataT>::value,
                           vec<convertT, NumElements>>::type
   convert() const {
+#ifdef __SYCL_DEVICE_ONLY__
```

Comment:
Why under `__SYCL_DEVICE_ONLY__`? We use clang's extended vector extension not only for the device but also for host under `__SYCL_USE_EXT_VECTOR_TYPE__` macro.

Reply:
Added comment with clarification.